packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +190,297 @@ class CupertinoAlertDialog extends StatelessWidget {
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container();
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return actionSection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        width: _kCupertinoDialogWidth,
+        padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+        child: ClipRRect(
+          borderRadius: const BorderRadius.all(const Radius.circular(_kDialogCornerRadius)),
+          child: new BackdropFilter(
+            filter: new ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+            child: new Container(
+              decoration: _kCupertinoDialogBlurOverlayDecoration,
+              child: new CupertinoDialogRenderWidget(
+                children: <Widget>[
+                  new CupertinoLayoutId(
+                    id: _AlertDialogSection.content,
+                    child: _buildContent(),
+                  ),
+                  new CupertinoLayoutId(
+                    id: _AlertDialogSection.actions,
+                    child: _buildActions(),
+                  ),
+                ],
+              ),
+            ),
+          ),
         ),
       ),
     );
   }
 }
 
+enum _AlertDialogSection {
+  content,
+  actions,
+}
+
+class CupertinoDialogRenderWidget extends MultiChildRenderObjectWidget {
+  CupertinoDialogRenderWidget({
+    Key key,
+    @required List<Widget> children,
+  }) : super(key: key, children: children);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new CupertinoDialogRenderBox();
+  }
+}
+
+class CupertinoDialogRenderBox extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, CupertinoDialogRenderBoxParentData>,
+        RenderBoxContainerDefaultsMixin<RenderBox, CupertinoDialogRenderBoxParentData> {
+  CupertinoDialogRenderBox({
+    List<RenderBox> children,
+  }) {
+    addAll(children);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! CupertinoDialogRenderBoxParentData)
+      child.parentData = new CupertinoDialogRenderBoxParentData();
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    RenderBox _content;
```

Comment:
This renderer is specific for the dialog. Just return _kCupertinoDialogWidth here and below (which should be true regardless of what the children are).